### PR TITLE
Add support for np.unwrap()

### PR DESCRIPTION
**Pull request by KrisMinchev**
This PR improves on #4074  by implementing the function `np.unwrap()`.

### CODE REVIEW

File: numba/np/arraymath.py [hunk 1]
```diff
@@ -4452,6 +4452,75 @@ def impl(a, dtype=None):
...
+
+    if not isinstance(discont, (types.Integer, types.Float, types.NoneType)):
+        msg = 'The argument "discont" must be a scalar'
+        raise TypingError(msg)
+
+    if not isinstance(period, (float, types.Number)):
+        msg = 'The argument "period" must be a scalar'
+        raise TypingError(msg)
+
+    axis = -1
```
[the remark this answers is not in the document]
Comment (KrisMinchev) [reply numba/np/arraymath.py:4474]: Hi @guilhermeleobas, I think that if the user wants to set `axis = -1` manually, such a check will result in an exception, which might not be the best way to handle that case.

File: numba/tests/test_np_functions.py [hunk 2]
```diff
@@ -441,6 +440,18 @@ def unwrap(p, discont=None, axis=-1, period=6.283185307179586):
     return np.unwrap(p, discont, axis, period=period)
 
 
+def unwrap1(p):
```
Comment (guilhermeleobas) [inline numba/tests/test_np_functions.py:443]: Just out of curiosity, is there any reasoning for the function naming convention?
* unwrap1
* unwrap13
* unwrap123

File: numba/np/arraymath.py [hunk 3]
```diff
@@ -4463,25 +4463,27 @@ def numpy_unwrap(p, discont=None, axis=-1, period=6.283185307179586):
         msg = 'The argument "p" must be array-like'
         raise TypingError(msg)
 
-    if not isinstance(discont, (types.Integer, types.Float, types.NoneType)):
+    if (not isinstance(discont, (types.Integer, types.Float, types.NoneType))
+            and (discont is not None)):
```
Comment (guilhermeleobas) [inline numba/np/arraymath.py:4467]: Use `cgutils.is_nonelike(discont)` to check if it is `None`

**Review by guilhermeleobas — CHANGES_REQUESTED** [review]
Just one last request and it should be good to merge

File: numba/tests/test_np_functions.py (hunk 2 above)
Comment (KrisMinchev) [reply numba/tests/test_np_functions.py:443]: This naming convention helps me keep track of which of `unwrap`'s arguments it needs as input(only first, first and third, etc.).

File: numba/np/arraymath.py [hunk 4]
```diff
@@ -4464,7 +4464,7 @@ def numpy_unwrap(p, discont=None, axis=-1, period=6.283185307179586):
         raise TypingError(msg)
 
     if (not isinstance(discont, (types.Integer, types.Float, types.NoneType))
-            and (discont is not None)):
+            and not cgutils.is_nonelike(discont)):
```
Comment (guilhermeleobas) [inline numba/np/arraymath.py:4467]: `cgutils.is_nonelike(discont)` already checks if discont is an instance of `types.NoneType`.

**Review by guilhermeleobas — APPROVED** [review]
LGTM. Thanks @KrisMinchev

**Comment by KrisMinchev** [comment]
Hi @guilhermeleobas, thank you for your review. I have refactored the implementation of `np.unwrap` accordingly.

**Review by guilhermeleobas — APPROVED** [review]
Thanks @KrisMinchev

**Pull request merged**